sys/net/gnrc/link_layer/lorawan/gnrc_lorawan_mcps.c

```diff
@@ -44,8 +44,25 @@ static int gnrc_lorawan_mic_is_valid(uint8_t *buf, size_t len, uint8_t *nwkskey)
     iolist_t iol =
     { .iol_base = buf, .iol_len = len - MIC_SIZE, .iol_next = NULL };
 
-    gnrc_lorawan_calculate_mic(&lw_hdr->addr, fcnt, GNRC_LORAWAN_DIR_DOWNLINK,
-                               &iol, nwkskey, &calc_mic);
+
+    /* for LoRaWAN 1.1 conf_fcnt is hardcoded to be 0 */
+    uint16_t conf_fnct = 0x00;
+
+    if (IS_ACTIVE(CONFIG_GNRC_LORAWAN_1_1)) {
+        /**
+        * TODO: If the device is connected to a LoRaWAN 1.1 Network Server and the
+        * ACK bit of the downlink frame is set, meaning this frame is acknowledging
+        * an uplink “confirmed” frame,then ConfFCnt is the frame counter value
+        * modulo 2^16 of the “confirmed” uplink frame that is being acknowledged.
+        * In all other cases ConfFCnt = 0x0000.
+        */
+        bool ack = lorawan_hdr_get_ack(lw_hdr);
```

Comment:
Could you implement that? Otherwise confirmed downlink message won't seem to work